protocol/dataunit/server.go

```diff
@@ -19,33 +35,44 @@ type Server struct {
 	writes  uint64
 	pending []transaction
 
-	Conn Conn
+	Conn io.ReadWriter
 }
 
-// ServeDataUnit reads one data unit from the client and provides a [Writer] to respond.
-// The returned Writer can only be called once. The returned Writer will always
+// ServeDataUnit reads one data unit from the client and provides a [Responder] to respond.
+//
+// The supplied Context must be non-nil, and only affects reading the request from the client.
+// Cancelling the Context after ServeDataUnit returns will have no effect on the Responder.
+//
+// The returned Responder can only be called once. The returned Responder will always
```

Comment:
Is there a way to enforce the "only called once" thing in code, perhaps with a `sync.Once` or similar so that multiple calls can return the original return value?

Reply:
Done.